### PR TITLE
docs: module page titles should not start with a link to themselves

### PR DESCRIPTION
**Pull request by nedbat**
The `:mod:` role used in the title of module pages turns into a link, but the link is to the same page.  That's distracting and unhelpful.  This uses the `!` prefix to instruct Sphinx to not make a link.

This was done with this thrown-together program:
```py
"""
Change every module page in the CPython docs 
so that the title doesn't link to itself.
"""

from pathlib import Path

# Edge cases:
# - modules with a blank line at the top.
# - modules that already had the !-prefix to suppress the link.
# - had to fix the rule length
# - pages that started with a link, but to the main module page.
#       unittest.mock-examples.rst
# - pages with slightly off names:
#       xml.sax.xmlreader is in xml.sax.reader.rst
#       xml.etree.ElementTree is in xml.etree.elementtree.rst
#       xml.parsers.expat is in pyexpat.rst
#       xml.sax.saxutils is in xml.sax.utils.rst
# - pages with two links in the title!
#       copyreg.rst

def fix_one_file(fpath):
    text = fpath.read_text().lstrip()
    if not text.startswith(":mod:`"):
        return
    if text.startswith(":mod:`!"):
        return
    lines = text.splitlines(keepends=True)
    modname = lines[0].split("`")[1]
    if modname != fpath.stem:
        print(f"WUT: {modname=}, {fpath.stem=}")
    lines[0] = lines[0].replace(":mod:`", ":mod:`!")
    lines[1] = lines[1][0] * (len(lines[0]) - 1) + "\n"
    fpath.write_text("".join(lines))
    print(f"Updated {fpath}")

for fname in Path(".").glob("**/*.rst"):
    fix_one_file(fname)
```

----
📚 Documentation preview 📚: https://cpython-previews--117099.org.readthedocs.build/

### CODE REVIEW

**Comment by AlexWaygood** [comment]
> This is only marked Draft because I didn't want to ping 29 other people for review. Can that be stopped?

I don't know of a way of avoiding that, unfortunately... It's not possible for me to deselect any of the codeowners from the list of people who are going to be requested for review.

File: Doc/library/__future__.rst
Comment (AlexWaygood) [inline Doc/library/__future__.rst]: It looks like in every page you've touched, you've added two characters to the "underline" line below the title, but you should have added only one

Comment (nedbat) [reply Doc/library/__future__.rst]: Ugh! Will fix.

Comment (nedbat) [reply Doc/library/__future__.rst]: Fixed.

File: Doc/library/copyreg.rst
Comment (AlexWaygood) [inline Doc/library/copyreg.rst:2]: The `pickle` link here _is_ useful, as it takes you to the documentation for another module:

```suggestion
:mod:`!copyreg` --- Register :mod:`pickle` support functions
============================================================
```

Comment (nedbat) [reply Doc/library/copyreg.rst:2]: Yes, this was one of the edge cases.  The pickle module is linked in the first paragraph, which is more useful in any case.

Comment (AlexWaygood) [reply Doc/library/copyreg.rst:2]: True. I'm okay with it just being linked in the first paragraph.

**Review by AlexWaygood — APPROVED** [review]
LGTM, thanks! I agree that these are distracting for readers, and serve no purpose.

While I agree with @CAM-Gerlach's comments on Discord that a Sphinx extension might be a cleaner way of fixing this, I'm personally happy to merge this unless somebody wants to volunteer to actually do that. I think we should prioritise the experience for readers of the docs, and this PR improves the experience of reading the docs.

**Comment by nedbat** [comment]
Since I don't want to ping 29 code owners, and am a little bit interested in learning about Sphinx internals, let's wait on this to see how the CAM+Ned tag team manages on tweaking Sphinx.

**Review by JacobCoffee — APPROVED** [review]

**Review by ezio-melotti — APPROVED** [review]

**Review by erlend-aasland — APPROVED** [review]

**Comment by nedbat** [comment]
Everyone: I know this might ping many people.  My apologies. The changes were auto-created across all of the module pages.

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

How we did not notice this earlier?

**Pull request merged**
